src/routes/contractViewState.js

```diff
@@ -0,0 +1,52 @@
+const warp = require('../warp');
+const { config } = require('../config');
+const { getLastStateFromDreCache, getCachedViewState, insertViewStateIntoCache } = require('../db/nodeDb');
+
+module.exports = async (ctx) => {
+  if (!config.availableFunctions.viewState) {
+    ctx.body = 'Contract view state functionality is disabled';
+    ctx.status = 503;
+    return;
+  }
+
+  const nodeDb = ctx.nodeDb;
+
+  try {
+    const contractId = ctx.query.id;
+    if (!contractId) {
+      throw new Error('Missing contract id');
```

Comment:
hmm, i would go with some more informative errors. Missing parameters seem to be more like a 422? Invalid tx id sounds like 400? With knex you can simply do
```
ctx.throw(422, 'Missing contract id.');
```

and then return it in the catch block like so:
```
ctx.status = e.status;
ctx.body = { message: e };
```

Reply:
Good point, I changed the http codes and the implementation.